src/PyNvCodec/src/PyNvEncoder.cpp

```diff
@@ -72,6 +72,24 @@ std::map<NV_ENC_CAPS, int> PyNvEncoder::Capabilities()
   return capabilities;
 }
 
+int PyNvEncoder::GetFrameSize() const 
+{ 
+     switch (GetPixelFormat()) {
+  case NV12:
+    return Width() * (Height() + (Height() + 1) / 2);
+  
+  case YUV444:
+    return Width() * Height() * 3;
+
+    case YUV444_10bit:
+    return 2 * Width() * Height() * 3;
+ 
+  default:
+    invalid_argument("Invalid Buffer format");
```

Comment:
You probably wanted to throw the execution (if this is std:: invalid_argument)

Reply:
ctor in invalid_argument() should take of that

Reply:
My understanding is that this will just construct the error value without it being thrown.

Reply:
you are right, "throw" was missing, added this now